demisto_sdk/commands/test_content/tests/timestamp_replacer_test.py

```diff
@@ -0,0 +1,265 @@
+        assert 'key1' not in time_stamp_replacer.query_keys
+
+    @pytest.mark.parametrize('time', TIMESTAMP_FORMATS)
+    def test_finding_problematic_keys_in_form_keys(self, mocker, flow, time):
+        """
+        Given:
+            - A timestamp replacer instance
+        When:
+            - searching a request for problematic keys that has timestamp values
+        Then:
+            - Ensure the problematic key is loaded into form_keys
+            - Ensure the valid key is not loaded into form_keys
+        """
+        mocker.patch('builtins.open', mock_open())
+        mitmproxy.ctx.options.detect_timestamps = True
+        mitmproxy.ctx.options.script_mode = 'record'
+        flow.request._set_urlencoded_form([('key1', 'value1'), ('timestamp_key', '2021-01-11T13:18:12+00:00')])
+        time_stamp_replacer = TimestampReplacer()
+        time_stamp_replacer.request(flow)
+        assert 'timestamp_key' in time_stamp_replacer.form_keys
+        assert 'key1' not in time_stamp_replacer.form_keys
+
+    @pytest.mark.parametrize('time', TIMESTAMP_FORMATS)
+    def test_finding_problematic_keys_in_json_keys(self, mocker, flow, time):
+        """
+        Given:
+            - A timestamp replacer instance
+        When:
+            - searching a request for problematic keys that has timestamp values
+        Then:
+            - Ensure the problematic key is loaded into json_keys
+            - Ensure the problematic key is loaded into json_keys even when it's nested
+            - Ensure the valid key is not loaded into json_keys
+        """
+        mocker.patch('builtins.open', mock_open())
+        mitmproxy.ctx.options.detect_timestamps = True
+        mitmproxy.ctx.options.script_mode = 'record'
+        flow.request.method = 'POST'
+        flow.request.set_content(json.dumps({'key1': 'value1',
+                                             'timestamp_key': '2021-01-11T13:18:12+00:00',
+                                             'dict1': {'list': ['test', '2021-01-11T13:18:12+00:00']}}).encode())
+        time_stamp_replacer = TimestampReplacer()
+        time_stamp_replacer.request(flow)
+        assert 'timestamp_key' in time_stamp_replacer.json_keys
+        assert 'dict1.list.1' in time_stamp_replacer.json_keys
+        assert 'key1' not in time_stamp_replacer.json_keys
+
+    def test_json_body_parsing(self, mocker, flow):
+        """
+        Given:
+            - A timestamp replacer instance
+        When:
+            - Trying to parse json body
+        Then:
+            - Ensure the body is parsed when is't json body
+            - Ensure the body is parsed when it's json body with single quotes
+        """
+        mocker.patch('builtins.open', mock_open())
+        mitmproxy.ctx.options.detect_timestamps = True
+        mitmproxy.ctx.options.script_mode = 'record'
+        flow.request.method = 'POST'
+        flow.request.set_content(str({'timestamp_key': '2021-01-11T13:18:12+00:00'}).encode())
```

Comment:
```suggestion
        flow.request.set_content(str({'timestamp_key': time}).encode())
```

Reply:
Not needed here - I'm testing different things here.

json timestamp parsing is tested in `test_finding_problematic_keys_in_json_keys`